opensrp-chw-core/src/main/java/org/smartregister/chw/core/interactor/BaseReferralNotificationDetailsInteractor.java

```diff
@@ -0,0 +1,93 @@
+package org.smartregister.chw.core.interactor;
+
+import android.app.Activity;
+import android.content.Context;
+
+import org.jetbrains.annotations.NotNull;
+import org.smartregister.chw.core.R;
+import org.smartregister.chw.core.contract.BaseReferralNotificationDetailsContract;
+import org.smartregister.chw.core.dao.ReferralNotificationDao;
+import org.smartregister.chw.core.domain.ReferralNotificationItem;
+import org.smartregister.chw.core.domain.ReferralNotificationRecord;
+import org.smartregister.family.util.Utils;
+import org.smartregister.opd.utils.OpdUtils;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class BaseReferralNotificationDetailsInteractor implements BaseReferralNotificationDetailsContract.Interactor {
+
+    private BaseReferralNotificationDetailsContract.Presenter presenter;
+    private Context context;
+
+    public BaseReferralNotificationDetailsInteractor(BaseReferralNotificationDetailsContract.Presenter presenter) {
+        this.presenter = presenter;
+        context = (Activity) presenter.getView();
+    }
+
+    @Override
+    public void fetchReferralDetails(String referralTaskId, String referralType) {
+        ReferralNotificationItem referralNotificationItem = null;
+
+        if (referralType.contains("Successful")) {
+            referralNotificationItem = getDetailsForSuccessfulReferral(referralTaskId);
+        } else if (referralType.contains("PNC") || referralType.contains("ANC")) {
+            referralNotificationItem = getDetailsForPNCAndANCReferrals();
+        } else if (referralType.contains("Malaria")) {
+            referralNotificationItem = getDetailsForMalariaReferral();
+        } else if (referralType.contains("Family Planning")) {
+            referralNotificationItem = getDetailsForFamilyPlanningReferral();
+        }
```

Comment:
Can we define the `referralTypes` as Constants? So that they're final and also easier to manage?

Reply:
Sure I have refactored this too.

Reply:
Extracted them to String resource

Reply:
Great 👍